content/ri/memory-optimizations.md

```diff
@@ -281,23 +278,22 @@ Suppose you are using large [sets](https://redis.io/docs/data-types/sets/) to so
 - Check if an observation already appeared in the stream.
 - Find the fraction or the number of observations in the stream that are smaller or larger than a given value.
 
-and you are ready to trade accuracy with speed and memory usage, consider using one of the following probabilistic data structures. You can use:
+If you are ready to trade accuracy with speed and memory usage, consider using one of the following probabilistic data structures. You can use:
 
 - [HyperLogLog](https://redis.io/docs/data-types/hyperloglogs/) to estimate the number of unique observations in a set.
 - [Bloom filter or cuckoo filter](https://redis.io/docs/stack/bloom/) to check if an observation has already appeared in the stream. False positive matches are possible, but false negatives are not.
 - [t-digest](https://redis.io/docs/stack/bloom/) to estimate the fraction or the number of observations in the stream that are smaller or larger than a given value.
 
 Bloom filter and cuckoo filter require [RedisBloom](https://redis.com/modules/redis-bloom/).
 
-#### Trade offs
+#### Trade-offs
 
-Following are the tradeoffs of using HyperLogLog:
+Following are the trade-offs of using HyperLogLog:
 
 1. The results achieved from HyperLogLog are not 100% accurate; they have an approximate standard error of 0.81%.
 1. Hyperloglog only tells you the unique count. It cannot tell you the elements in the set.
```

Comment:
can be used only for estimating the count of unique observations, and not for checking if a specific value was observed.